app/src/main/java/io/gnosis/safe/ui/base/BaseFragment.kt

```diff
@@ -1,51 +1,41 @@
 package io.gnosis.safe.ui.base
 
 import android.content.Context
-import android.os.Bundle
-import android.view.LayoutInflater
-import android.view.View
-import android.view.ViewGroup
 import androidx.fragment.app.Fragment
-import androidx.viewbinding.ViewBinding
 import io.gnosis.safe.HeimdallApplication
+import io.gnosis.safe.ScreenId
+import io.gnosis.safe.Tracker
 import io.gnosis.safe.di.components.DaggerViewComponent
 import io.gnosis.safe.di.components.ViewComponent
 import io.gnosis.safe.di.modules.ViewModule
+import javax.inject.Inject
 
-abstract class BaseFragment<T> : Fragment()
-        where T : ViewBinding {
+abstract class BaseFragment : Fragment() {
 
-    private var _binding: T? = null
-    protected val binding get() = _binding!!
+    @Inject
+    lateinit var tracker: Tracker
 
     override fun onAttach(context: Context) {
         super.onAttach(context)
-        inject(buildViewComponent(context))
+        HeimdallApplication[requireContext()].inject(this)
```

Comment:
This returns the `appComponent` not the `viewComponent`. I think the `appComponent` is passed to the `viewComponent` when you are building it. I am not sure this necessary to call here.

Reply:
tracker is injected from app component

Reply:
yes, but the fragment receives its dependencies from the `viewComponent` which itself should provide any dependencies from the appComponent automatically as far as I understand. You shouldn't have to call `inject` twice in different components

Reply:
this is also exactly how we do it for our activities now. Same approach we allready have in our codebase. I don't think this refactoring is in scope of this pr

Reply:
Also injection should happen exactly where @Inject annotation is specified. So current approach might be the only way. You can't inject specific view models into base class